prow/plugins/assign/assign_test.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package assign
+
+import (
+	"testing"
+
+	"github.com/Sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+)
+
+type fakeClient struct {
+	assigned   []string
+	unassigned []string
+}
+
+func (c *fakeClient) UnassignIssue(owner, repo string, number int, assignees []string) error {
+	c.unassigned = assignees
+	return nil
+}
+
+func (c *fakeClient) AssignIssue(owner, repo string, number int, assignees []string) error {
+	c.assigned = assignees
+	return nil
+}
+
+func TestParseLogins(t *testing.T) {
+	var testcases = []struct {
+		name   string
+		text   string
+		logins []string
+	}{
+		{
+			name: "empty",
+			text: "",
+		},
+		{
+			name:   "one",
+			text:   " jungle",
+			logins: []string{"jungle"},
+		},
+		{
+			name:   "two",
+			text:   " erick fejta",
+			logins: []string{"erick", "fejta"},
+		},
+	}
+	for _, tc := range testcases {
+		l := parseLogins(tc.text)
+		if len(l) != len(tc.logins) {
+			t.Errorf("For case %s, expected %s and got %s", tc.name, tc.logins, l)
+		}
+		for n, who := range l {
+			if tc.logins[n] != who {
+				t.Errorf("For case %s, expected %s and got %s", tc.name, tc.logins, l)
+			}
+		}
+	}
+}
+
+func TestAssignComment(t *testing.T) {
+	var testcases = []struct {
+		name      string
+		action    string
+		body      string
+		commenter string
+		added     []string
+		removed   []string
+	}{
+		{
+			name:      "unrelated comment",
+			action:    "created",
+			body:      "uh oh",
+			commenter: "o",
+		},
+		{
+			name:      "not created",
+			action:    "something",
+			body:      "uh oh",
+			commenter: "o",
+		},
+		{
+			name:      "assign me",
+			action:    "created",
+			body:      "/assign",
+			commenter: "rando",
+			added:     []string{"rando"},
+		},
+		{
+			name:      "unassign myself",
+			action:    "created",
+			body:      "/unassign",
+			commenter: "rando",
+			removed:   []string{"rando"},
+		},
+		{
+			name:      "assign friends",
+			action:    "created",
+			body:      "/assign bert ernie",
+			commenter: "rando",
+			added:     []string{"bert", "ernie"},
+		},
+		{
+			name:      "unassign buddies",
+			action:    "created",
+			body:      "/unassign ashitaka eboshi",
+			commenter: "san",
+			removed:   []string{"ashitaka", "eboshi"},
+		},
```

Comment:
I would maybe test more controversial possibilities: inconsistent spaces, inconsistent characters (dash seems to be the only one accepted, what about @?)